merlin/dag/graph.py

```diff
@@ -36,23 +36,22 @@ class Graph:
     transforms dataframes or dataframe-like data
     """
 
-    def __init__(self, output_node: Node, subgraphs: Optional[Dict[str, Node]] = None):
+    def __init__(self, output_node: Node):
         self.output_node = output_node
-        self.subgraphs = subgraphs or {}
+        self.subgraphs = {}
 
         parents_with_deps = self.output_node.parents_with_dependencies
         parents_with_deps.append(output_node)
 
-        for name, sg in self.subgraphs.items():
-            if sg not in parents_with_deps:
-                raise ValueError(
-                    f"The output node of subgraph {name} does not exist in the provided graph."
-                )
+        for node in postorder_iter_nodes(output_node):
+            op = node.op
+            if op.is_subgraph and op.name:
```

Comment:
The `is_subgraph` property is a workaround to avoid the circular import issue that would result from making `Graph` depend on `Subgraph` in order to do an `isinstance` check on this line